src/Dropdown/index.jsx

```diff
@@ -1,225 +1,169 @@
-            'has-icon': hasIconElement,
-            rounded: hasIconElement,
-            border: hasIconElement,
-            'd-flex': hasIconElement,
-            'bg-white': hasIconElement,
-          },
-        )}
+        className={this.state.open ? 'dropdown show' : 'dropdown'}
         ref={(container) => { this.container = container; }}
       >
-        { hasIconElement &&
-          <div
-            className={classNames(
-              'icon-container',
-              'd-flex',
-              'align-items-center',
-              'justify-content-center',
-              'border-right',
-            )}
-          >
-            {React.cloneElement(iconElement, {
-              className: iconElement.props ? classNames(iconElement.props.className, 'rounded-left') : null,
-            })}
-          </div>
-        }
         <Button
+          {...other}
           aria-expanded={open}
           aria-haspopup="true"
-          buttonType={buttonType}
+          className={buttonClassName}
           onClick={this.toggle}
-          onKeyDown={this.handleToggleKeyDown}
-          className={classNames(
-            'dropdown-toggle',
-            {
-              'border-0': hasIconElement,
-              'rounded-0': hasIconElement,
-              'bg-white': hasIconElement,
-            },
-          )}
           type="button"
-          inputRef={(toggleElem) => { this.toggleElem = toggleElem; }}
+          inputRef={(toggleButton) => { this.toggleButton = toggleButton; }}
         >
-          {title}
+          {buttonContent}
         </Button>
-        <div
-          aria-label={title}
-          aria-hidden={!open}
-          className={classNames(
-            'dropdown-menu',
-            { show: open },
-          )}
-          role="menu"
-        >
-          {this.generateMenuItems(menuItems)}
+        {/* eslint-disable-next-line jsx-a11y/interactive-supports-focus */}
+        <div className={this.state.open ? 'dropdown-menu show' : 'dropdown-menu'} aria-label={buttonContent} aria-hidden={!open} role="menu" ref={this.menuItems} onKeyDown={this.handleMenuKeyDown}>
+          {this.props.children}
         </div>
       </div>
     );
   }
 }
 
 Dropdown.propTypes = {
-  buttonType: PropTypes.string,
-  iconElement: PropTypes.element,
-  menuItems: PropTypes.arrayOf(
-    PropTypes.shape({
-      label: PropTypes.string,
-      href: PropTypes.string,
-    }),
-    PropTypes.element,
-  ).isRequired,
-  title: PropTypes.string.isRequired,
+  buttonClassName: PropTypes.string,
+  children: PropTypes.node.isRequired,
+  buttonContent: PropTypes.string.isRequired,
 };
 
 Dropdown.defaultProps = {
-  buttonType: 'light',
-  iconElement: undefined,
+  buttonClassName: 'dropdown-toggle',
 };
 
 export default Dropdown;
+
+Dropdown.Item = props => {
+  const { type, children, ...other } = props;
+  const item = React.createElement(
+    type,
+    {
+      ...other,
+      className: 'dropdown-item',
```

Comment:
A consuming application should be able to specify a `className` prop to a DropdownItem. Let's update this line to append the 'dropdown-item' class to a className prop if it's supplied.

Reply:
Should 'dropdown-item' be set as defaultProps?

Reply:
No let's leave this one baked in.